src/inmanta/agent/agent.py

```diff
@@ -599,6 +561,25 @@ async def mark_deployment_as_finished(self, resource_actions: Iterable[ResourceA
                 self.agent.process.add_background_task(self.agent.get_latest_version_for_agent(self.deferred))
                 self.deferred = None
 
+    async def resolve_cads(self, version: int, cads: dict[ResourceIdStr, RemoteResourceAction]) -> None:
+        """ Batch resolve all CAD's """
+        async with self.agent.process.cad_ratelimiter:
+            result = await self.get_client().resources_status(
+                tid=self.agent._env_id,
+                model_version=version,
+                rids=list(cads.keys()),
+            )
+            if result.code != 200 or result.result is None:
+                LOGGER.error("Failed to get the status for remote resources %s (%s)", list(cads.keys()), result.result)
+                return
+
+            for rid_raw, status_raw in result.result["data"].items():
+                status = const.ResourceState[status_raw]
+                rra = cads.get(rid_raw, None)
```

Comment:
Is there a specific scenario where a resource id would be returned by `resources_status` but not be present in the `cads` dict ?

Reply:
mhh, right. That shouldn't happen (the other way around can happen)